src/export/StructureDefinitionExporter.ts

```diff
@@ -482,6 +482,20 @@ export class StructureDefinitionExporter implements Fishable {
           // Note: newElement() method automatically adds the new element to its structDef.elements
           const newElement = structDef.newElement(rule.path);
           newElement.applyAddElementRule(rule, this);
+          const lastIdPart = newElement.id.slice(newElement.id.lastIndexOf('.') + 1);
+          // see http://hl7.org/fhir/elementdefinition-definitions.html#ElementDefinition eld-19 and eld-20
+          const nameErrors: string[] = [];
+          if (!/^[^\s\\.,:;\\'"\/|?!@#$%&*()\[\]{}]+(\[x\])?$/.test(lastIdPart)) {
+            nameErrors.push('cannot include some special characters');
+          }
+          if (lastIdPart.length > 64) {
+            nameErrors.push('must be at most 64 characters long');
+          }
+          if (nameErrors.length > 0) {
+            logger.error(`Element names ${nameErrors.join(' and ')}.`, rule.sourceInfo);
+          } else if (!/^[A-Za-z][A-Za-z0-9]*$/.test(lastIdPart)) {
+            logger.warn('Element names should be simple alphanumerics.', rule.sourceInfo);
+          }
```

Comment:
I like the simple messages, but I think users would benefit from a bit more... I'd suggest you add something like `See eld-19 at https://hl7.org/fhir/elementdefinition.html#ElementDefinition-inv.` (and likewise for eld-20).

I also wonder if we should add the violating path to the message?

E.g.
> Invalid path: Foo.b@r. Element names cannot include some special characters. See eld-19 at https://hl7.org/fhir/elementdefinition.html#ElementDefinition-inv.

and:
> Unadvised path: Foo.bar_stool. Element names should be simple alphanumerics. See eld-20 at https://hl7.org/fhir/elementdefinition.html#ElementDefinition-inv.

(I don't love the word "unadvised" but having trouble coming up w/ a good word for not-recommended).

Reply:
I think I'll go with "inadvisable," that's about the best I can think of.

Reply:
I would have said `0/10 recommended path: Foo.bar_stool` but I like your choice better 😆